### PR TITLE
Fixed tests - probably terribly

### PR DESCRIPTION
**Pull request by psav**
This is a starting point to fixing the tests - it was helpful for me to understand a little more about how receptor is working internally, please advise on pieces I should modify.

All 5 tests now pass, but I'm sure there are some items wrong in this PR.

### CODE REVIEW

File: test/integration/test_main.py [hunk 1]
```diff
@@ -22,20 +20,23 @@ def receptor_service(receptor_config):
...
-        config.setdefault('server', {})['port'] = unused_tcp_port_factory()
-        config['peers'] = {'127.0.0.1:{}'.format(p): '' for p in peer_ports}
-        receptor_config = ReceptorConfig(cmdline_args=config)
-        return Receptor(config=receptor_config, node_id=node_name)
+    def _receptor_service(node_name, peer_ports=[], type="node"):
+        peers = {'127.0.0.1:{}'.format(p): '' for p in peer_ports}
+        peer_config = []
+        for peer in peers:
+            peer_config.extend(['--peer', peer])
+        base_config = ['--node-id', node_name, '--data-dir', '/tmp/receptor', type, '--listen-port', str(unused_tcp_port_factory())]
```
Comment (matburt) [inline test/integration/test_main.py]: This bit here implies that `/tmp/receptor` already exists, either we need to pre-create it or have the test runner create a tmp dir using `mkdtemp`?

Comment (psav) [reply test/integration/test_main.py]: Ahh yes - I can fix that if the rest of the implementation seems OK

Comment (matburt) [reply test/integration/test_main.py]: Yep the rest looks good to me!

Comment (matburt) [reply test/integration/test_main.py]: Although... no self deprecating comments in commit messages ;)

Comment (psav) [reply test/integration/test_main.py]: awww man!!

**Review by matburt — APPROVED** [review]

**Comment by matburt** [comment]
I do get some messages indicating we're probably not handling some tasks but I think that's unrelated to this and we'll need to rework some more of our core unit testing to fix them

**Pull request merged**